Bug-Ubuntu: https://bugs.launchpad.net/bugs/1986852
Origin: upstream, https://git.savannah.nongnu.org/gitweb/?p=dmidecode.git;a=commit;h=024b0fb43ada238659c61f8d2e1f9ddeae603708
From: Coiby Xu <coxu@redhat.com>
Date: Thu, 21 Oct 2021 15:19:06 +0200
Subject: [PATCH] Support new byte value of Processor Upgrade field from SMBIOS
 3.5.0

Intel(R) Xeon(R) Platinum 8375C CPU @ 2.90GHz has byte value 3Fh which
means Socket LGA4677 for Processor Information - Processor Upgrade
field.

[JD: Fix upper array boundary check.]

Signed-off-by: Coiby Xu <coxu@redhat.com>
Signed-off-by: Jean Delvare <jdelvare@suse.de>
---
 dmidecode.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/dmidecode.c b/dmidecode.c
index 0b44182de5d5..3147d4dd6786 100644
--- a/dmidecode.c
+++ b/dmidecode.c
@@ -1394,10 +1394,11 @@ static const char *dmi_processor_upgrade(u8 code)
 		"Socket BGA1510",
 		"Socket BGA1528",
 		"Socket LGA4189",
-		"Socket LGA1200" /* 0x3E */
+		"Socket LGA1200",
+		"Socket LGA4677" /* 0x3F */
 	};
 
-	if (code >= 0x01 && code <= 0x3E)
+	if (code >= 0x01 && code <= 0x3F)
 		return upgrade[code - 0x01];
 	return out_of_spec;
 }
-- 
2.34.1

